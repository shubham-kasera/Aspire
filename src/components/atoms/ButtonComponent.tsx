import React, {FC} from 'react';
import {StyleSheet, StyleProp, ViewStyle} from 'react-native';
import {Button} from 'react-native-paper';
import {hp, wp} from '../../styles/globalStyles';
import {size, type} from '../../theme/fonts';

interface Button {
  title: string;
  disabled?: boolean;
  buttonStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<ViewStyle>;
  onPress?: () => void;
}

const ButtonComponent: FC<Button> = props => {
  return (
    <Button
      mode="contained"
      style={[
        styles.buttonStyle,
        props.buttonStyle,
        props.disabled ? {backgroundColor: '#EEE'} : null,
      ]}
      labelStyle={[
        styles.labelStyle,
        props.labelStyle,
        props.disabled ? {color: '#FFF'} : null,
      ]}
      disabled={props.disabled}
      onPress={props.onPress}
      contentStyle={{paddingVertical: '4%'}}>
      {props.title}
    </Button>
  );
};

const styles = StyleSheet.create({
  buttonStyle: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: hp('4%'),
    width: wp('70%'),
    borderRadius: 50,
    backgroundColor: '#01D167',
  },
  labelStyle: {
    fontSize: size.font16,
    textTransform: 'capitalize',
    fontFamily: type.avenirDemi,
  },
});

export default ButtonComponent;
