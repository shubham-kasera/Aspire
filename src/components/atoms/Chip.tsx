import React, {FC} from 'react';
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  StyleProp,
  ViewStyle,
} from 'react-native';
import {hp} from '../../styles/globalStyles';
import {size, type} from '../../theme/fonts';

interface Chip {
  title: string;
  wrapperStyle?: StyleProp<ViewStyle>;
  titleStyle?: StyleProp<ViewStyle>;
  onPress?: () => void;
}

const Chip: FC<Chip> = props => {
  return (
    <TouchableOpacity
      activeOpacity={1}
      style={[styles.ChipWrapper, props.wrapperStyle]}
      onPress={props.onPress}>
      <Text style={[styles.title, props.titleStyle]}>{props.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  ChipWrapper: {
    backgroundColor: '#01D167',
    paddingVertical: hp('0.5%'),
    paddingHorizontal: hp('1.8%'),
    borderRadius: 4,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: '#fff',
    fontSize: size.font12,
    fontFamily: type.avenirBold,
  },
});

export default Chip;
