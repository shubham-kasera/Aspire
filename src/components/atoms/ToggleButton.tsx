import React, {FC} from 'react';
import {
  StyleSheet,
  StyleProp,
  ViewStyle,
  TouchableOpacity,
  Text,
  Platform,
} from 'react-native';
import {Button} from 'react-native-paper';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {flexRow, hp} from '../../styles/globalStyles';
import {size, type} from '../../theme/fonts';

interface Button {
  title: string;
  icon: string;
  buttonStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<ViewStyle>;
  onPress?: () => void;
}

const ToggleButton: FC<Button> = props => {
  return (
    <TouchableOpacity
      activeOpacity={1}
      style={[flexRow.between, styles.showCardWrapper, props.buttonStyle]}
      onPress={props.onPress}>
      <MaterialIcons name={props.icon} color="#01D167" size={hp('2.70%')} />
      <Text style={[styles.showCardText, props.labelStyle]}>{props.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  showCardWrapper: {
    alignSelf: 'flex-end',
    backgroundColor: '#fff',
    paddingHorizontal: '3%',
    paddingTop: 7,
    paddingBottom: Platform.OS == 'ios' ? hp('3%') : 30,
    borderRadius: 6,
    marginBottom: -22,
    width: '45%',
  },
  showCardText: {
    color: '#01D167',
    fontSize: size.font12,
    fontFamily: type.avenirDemi,
    marginLeft: '2%',
  },
});

export default ToggleButton;
