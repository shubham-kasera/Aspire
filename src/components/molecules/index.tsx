import Header from './Header';
import BottomSheet from './BottomSheet';
import CardComponent from './CardComponent';
import IconListItem from './IconListItem';
import ProgressBarComponent from './ProgressBarComponent';

export {Header, BottomSheet, CardComponent, IconListItem, ProgressBarComponent};
