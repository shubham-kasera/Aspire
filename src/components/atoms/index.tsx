import Chip from './Chip';
import ButtonComponent from './ButtonComponent';
import ToggleButton from './ToggleButton';

export {Chip, ButtonComponent, ToggleButton};
