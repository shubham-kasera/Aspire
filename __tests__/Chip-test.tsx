/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Chip} from '../src/components/atoms';

jest.mock('react-native-gesture-handler', () => {});

it('renders correctly', () => {
  const tree = renderer.create(<Chip title="This is chip" />).toJSON();
  expect(tree).toMatchSnapshot();
});
