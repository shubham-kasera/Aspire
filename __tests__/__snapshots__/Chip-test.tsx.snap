// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#01D167",
      "borderRadius": 4,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 6.5,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "AvenirNextLTPro-Bold",
          "fontSize": 17.5,
        },
        undefined,
      ]
    }
  >
    This is chip
  </Text>
</View>
`;
